contracts/payment/SponsorPayment.sol

```diff
@@ -0,0 +1,519 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import {ReentrancyGuardUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol";
+import {EIP712Upgradeable} from "@openzeppelin/contracts-upgradeable/utils/cryptography/EIP712Upgradeable.sol";
+import {Ownable2StepUpgradeable} from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title SponsorPayment
+ * @dev A contract for sponsored payments between Sponsors and Recipient$.
+ *      Supports Ether and ERC-20 token payments with enhanced security feature$.
+ * @custom:storage-location erc7201:iden3.storage.SponsorPayment
+ */
+contract SponsorPayment is ReentrancyGuardUpgradeable, EIP712Upgradeable, Ownable2StepUpgradeable {
```

Comment:
I want to ask @AndriianChestnykh if we inherit different upgradeable contracts and they are all Initializable ( have the same storage location), will be that a problem for upgrade?